.github/workflows/cml.yaml

```diff
@@ -1,52 +1,35 @@
 name: train-my-model
-
 on: [push]
-
 jobs:
   run:
     runs-on: [ubuntu-latest]
-  
     steps:
       - uses: actions/checkout@v2
-
       - uses: iterative/setup-cml@v1
-
-      - name: cml_run
-        env:
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install
         run: |
+          cml ci  # complete repository setup for CML
           pip install -r requirements.txt
-
-          # Reproduce pipeline if any changes detected in dependencies
-          dvc repro
-
+      - name: Reproduce pipeline if any changes detected in dependencies
+        run: dvc repro
+      - name: CML report
+        env:
+          REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
           # Publish confusion matrix diff
-          echo -e "## Plots\n### Class confusions" >> report.md
+          echo "## Plots" > report.md
+          echo "### Class confusions" >> report.md
           dvc plots diff \
             --target output/predictions.json \
             --show-vega > vega.json
-          vl2png vega.json -s 1.5 | cml-publish --md >> report.md
-          # Add figure to report
-          cat output/metrics.json >> report.md
-          cml-send-comment report.md
-
-      - name: Commit files
-        id: commit
+          vl2png vega.json -s 1.5 | cml publish --md >> report.md
+          echo "## Metrics" >> report.md
+          dvc metrics show --md >> report.md
+          cml send-comment report.md
+      - name: Push changes
         run: |
-          git config --local user.email "action@github.com"
-          git config --local user.name "github-actions"
           git add --all
-          if [-z "$(git status --porcelain)"]; then
-            echo "::set-output name=push::false"
-          else
-            git commit -m "Results of the last experiment" -a
-            echo "::set-output name=push::true"
+          if [[ -n "$(git status --porcelain)" ]]; then
+            git commit -m "Results of the last experiment"
+            git push origin HEAD:${{ github.ref }}
           fi
```

Comment:
alternatively could (should?) use `cml pr .` here /CC @iterative/cml

Reply:
With https://github.com/iterative/cml/issues/907, `cml pr --auto-merge .` would be an awesome alternative to all this `git` trickery.

Reply:
Update: now `cml pr --squash .` as per https://github.com/iterative/cml/issues/823 and children